2024-11-07 01:29:25: Experiment log path in: /workspaces/ST-SSL/experiments/NYCBike1/20241107-012925
2024-11-07 01:29:25: Experiment configs are: Namespace(batch_size=32, best_path='None', d_input=2, d_model=64, d_output=2, data_dir='data', dataset='NYCBike1', debug=False, device='cpu', dropout=0.2, early_stop=True, early_stop_patience=15, epochs=100, grad_norm=True, graph_file='data/NYCBike1/adj_mx.npz', input_length=19, log_dir='/workspaces/ST-SSL/experiments/NYCBike1/20241107-012925', lr_init=0.001, max_grad_norm=5, mode='train', nmb_prototype=6, num_nodes=128, percent=0.1, seed=31, shm_temp=0.5, temp=4, test_batch_size=32, use_dwa=True, yita=0.6)
2024-11-07 01:29:26: Traceback (most recent call last):
  File "main.py", line 61, in model_supervisor
    results = trainer.train() # best_eval_loss, best_epoch
  File "/workspaces/ST-SSL/model/trainer.py", line 110, in train
    train_epoch_loss, loss_t = self.train_epoch(epoch, loss_weights)
  File "/workspaces/ST-SSL/model/trainer.py", line 58, in train_epoch
    loss, sep_loss = self.model.loss(repr1, repr2, target, self.scaler, loss_weights)
  File "/workspaces/ST-SSL/model/models.py", line 66, in loss
    l2 = self.temporal_loss(z1, z2)
  File "/workspaces/ST-SSL/model/models.py", line 84, in temporal_loss
    return self.thm(z1, z2)
  File "/opt/conda/envs/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/workspaces/ST-SSL/model/layers.py", line 113, in forward
    loss = self.b_xent(logits, self.lbl)
  File "/opt/conda/envs/py38/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1177, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'TemporalHeteroModel' object has no attribute 'lbl'

